experimental/PropertyDDS/packages/property-properties/src/properties/nodeProperty.js

```diff
@@ -2,385 +2,40 @@
-    this._insert(in_property.getId(), in_property, true);
-};
-
-/**
- * @override
- * @inheritdoc
- */
-NodeProperty.prototype._getScope = function () {
-    if (this._parent) {
-        return this.getRoot()._getScope();
-    } else {
-        if (this._checkedOutRepositoryInfo) {
-            return this._checkedOutRepositoryInfo.getScope();
-        } else {
-            return undefined;
-        }
-    }
-};
-
-/**
- * Removes the given property
- *
- * @param {string|property-properties.BaseProperty} in_property - The property to remove
- *                                                          (either its id or the whole property).
- * @throws if trying to remove an entry that does not exist
- * @return {property-properties.BaseProperty} the property removed.
- */
-NodeProperty.prototype.remove = function (in_property) {
-    var id = in_property;
-    var returnValue;
-    if (id instanceof BaseProperty) {
-        returnValue = id;
-        id = id.getId();
-    } else {
-        returnValue = this.get(id);
-    }
-    this._removeByKey(id);
-    return returnValue;
-};
-
-/**
- * Removes all dynamic children
- * @throws if node property is read-only
- */
-NodeProperty.prototype.clear = function () {
-    this._checkIsNotReadOnly(true);
-    _.each(this._entries, this.remove.bind(this));
-};
-
-/**
- * Inserts a property into the collection
- *
- * @param {string}                      in_key      -
- *     Key of the entry in the collection
- * @param {property-properties.NamedProperty} in_property -
- *     The property to insert
- * @param {boolean}                     in_reportToView -
- *     By default, the dirtying will always be reported to the checkout view and trigger a modified event there.
- *     When batching updates, this can be prevented via this flag.
- */
-NodeProperty.prototype._insert = function (in_key, in_property, in_reportToView) {
-    this._checkIsNotReadOnly(true);
-
-    // Add the child property to the dynamic properties
-    IndexedCollectionBaseProperty.prototype._insert.call(this, in_key, in_property, false);
-
-    if (this._typeid !== 'NodeProperty') {
-        // If this is not a NodeProperty (where children and entries are the same), we insert
-        // it into the children
-        this._append(in_property, false);
-    }
-
-    // We postponed the report above, to make sure the child property has actually been appended to this
-    // node, before the report is forwarded to the view
-    if (in_reportToView) {
-        this._reportDirtinessToView();
-    }
-};
 
 /**
- * Removes an entry with the given key
- *
- * @param {string} in_key - key of the entry
- * @param {boolean} in_reportToView -
- *     By default, the dirtying will always be reported to the checkout view and trigger a modified event there.
- *     When batching updates, this can be prevented via this flag.
- */
-NodeProperty.prototype._removeByKey = function (in_key, in_reportToView) {
```

Comment:
It seems like many of the logic from the file is moved to ContainerProperty too, I would suggest to manually and carefully review the changes so we don't miss any changes.